autopush/endpoint.py

```diff
@@ -867,3 +868,69 @@ def _load_params(self):
             return json.loads(self.request.body)
         except ValueError:
             return {}
+
+
+class BridgeMessageHandler(RegistrationHandler):
+    cors_methods = "PUT"
+
+    @cyclone.web.asynchronous
+    def put(self, uaid):
+        """HTTP PUT
+
+        Process message states for a UAID
+
+        """
+        self.start_time = time.time()
+        if not self._validate_auth(uaid):
+            return self._write_unauthorized_response(
+                message="Invalid Authentication")
+        params = self._load_params()
+        if 'nak' in params:
+            naks = params['nak']
+            batch = []
+            for nak in naks:
```

Comment:
It might be slightly more efficient to decrypt everything on a single thread, instead of punting to the pool for every nack. Something like `deferToThread(map, lambda msg_id: self.ap_settings.fernet.decrypt(msg_id.encode('utf8')), naks)`. Then again, we're sending so few nacks that it probably doesn't matter.

:bike: :art: :house: (I ❤️ this emoji).